shapely/coordinates.py

```diff
@@ -129,6 +142,127 @@ def transform(
     return result
 
 
+def transform_resize(
+    geom: Optional[shapely.Geometry],
+    transformation,
+    include_z: Optional[bool] = False,
+    interleaved: bool = True,
+):
+    """Apply a function to the coordinates of a Geometry object, optionally changing the number
+    of coordinate pairs.
+
+    This function differs with `transform` in the following ways:
+
+    - The number of coordinates per Geometry is allowed to change.
+    - The input parameters to the transformation function can also be (scalar) floats.
+    - It only accepts scalar Geometry objects, not arrays.
+
+    We recommend only using this function only if 1) the number of coordinates in your Geometry
+    should change or 2) the transformation can't be made to accept numpy arrays.
```

Comment:
I think another defining characteristic of this version is that the function is applied _per coordinate sequence_ (ring), instead of applied to all coordinates at once?